libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -85,6 +77,71 @@ std::unique_ptr<CFG> ControlFlowGraphBuilder::build(
 		cxx20::erase_if(node->entries, [&](CFG::BasicBlock* entry) -> bool {
 			return !reachabilityCheck.visited.count(entry);
 		});
+}
+// Sets the ``recursive`` member to ``true`` for all recursive function calls.
+void markRecursiveCalls(CFG& _cfg)
```

Comment:
This is different from the call graph generator and the recursive function finder in the yul optimizer in two ways:
- It works on non-disambiguated ``Scope::Function``s.
- It determines which *calls* are recursive rather than which *functions* are recursive.

Both of them combined justify keeping this separate, I'd say.